gh-132983: Add documentation for compression.zstd (#133911)
python/cpython: Doc/library/compression.rst

The :mod:`!compression` package
===============================

> 

Reviewer: Can we have a source code link like all other library pages.

Reviewer 2: Not needed, the corollary is `concurrent`.

.. versionadded:: 3.14

> .. attention::

Reviewer: This does not need the directive IMO

Reviewer 2: I saw it as a transitional thing, but happy to remove or keep

Author: I think it is useful and important to draw attention to this notice which, as Adam said, is transitional. We want to make sure users are aware that these names are the new canonical names.

Reviewer: ![image](https://github.com/user-attachments/assets/2ba23866-ba8f-43cd-819b-bec4833d7844)

I don't have a strong preference against, but IMO I still see little value.

>    The :mod:`!compression` package is the new location for the data compression

Reviewer 3: "New" always seems such a temporal term.  I'm just spitballing here, but how about something like: "The compression package provides a unified interface to several data compression routines. Some of these have historically been available as separate modules; those will continue to be available under their original names for compatibility reasons, and will not be removed without a deprecation cycle".

Author: Hm, I definitely understand "new" may not be the right word here. I worry if we say "a unified interface to ... compression routines" that people will expect `compression` to hold some high level `compress` function that abstracts over the various implementations, which is not the case.

Perhaps
"The compression package contains the canonical compression modules containing interfaces to several different algorithms. Some of these have historically been available as separate modules; those will continue to be available under their original names for compatibility reasons, and will not be removed without a deprecation cycle. The compression.* import names are encouraged for use where practical."

Author: I pushed up a slightly tweaked version of my suggestion. Please let me know what you think!

Reviewer 3: the intro looks good

   modules in the standard library, listed below. The existing modules are not
   deprecated and will not be removed before Python 3.19. The new ``compression.*``

>    import names are encouraged for use where practicable.

Reviewer 3: Just "practical".

Reviewer: *practicable* is grammatically correct, no? I think it’s fine.

Reviewer 4: Both are grammatical with slightly different meanings. "Practical" is more direct clearer.

Practicable: "are encouraged for use where [it could generally be done]"

Practical: "are encouraged for use where [it can actually be done]"

```suggestion
   import names are encouraged for use where practical.
```

Reviewer 2: I suggested practicable as I think it covers the intent more closely, compare *Able to be done or put into practice successfully; feasible* (“Practicable, Adj., Sense 1.”) with *Of, relating to practice or action, as opposed to speculation or theory.* (“Practical, Adj., Sense I.1.a.”), both copyright the Oxford English Dictionary. If both Hugo and Mats prefer changing, though, I have no strong objection.

Reviewer 3: I certainly won't raise objections if it stays. I've spent years trying to convince myself to use the "simpler" word in docs when there are a couple of choices, is all it was.

> :mod:`!compression` is encouraged where practical.

Reviewer 5: I think it would be better for the documentation of the four original modules to reflect their new preferred names. That way people reading e.g. the _gzip_ documentation will be aware of the new preferred module name.

Author: I think that is something we can add in the future, but not something I want to cover in this PR. I think the name change is an important piece of information people reading the compression module docs would want to know.

> The :mod:`!compression` package
> =================================

Reviewer 2: ```suggestion
The :mod:`!compression` package
===============================
```

.. versionadded:: 3.14

> .. note::
> 
>     Several modules in :mod:`!compression` re-export modules that currently
>     exist at the repository top-level. These re-exported modules are the new
>     canonical import name for the respective top-level module. The existing
>     modules are not currently deprecated and will not be removed prior to Python
>     3.19, but users are encouraged to migrate to the new import names when
>     feasible.

Reviewer 2: ```suggestion
.. attention::

   The :mod:`!compression` package is the new location for the data compression
   modules in the standard library, listed below. The existing modules are not
   deprecated and will not be removed before Python 3.19. The new ``compression.*``
   import names are encouraged for use where practicable.
```

> * :mod:`!compression.bz2` -- Re-exports :mod:`bz2`
> 
> * :mod:`!compression.gzip` -- Re-exports :mod:`gzip`
> 
> * :mod:`!compression.lzma` -- Re-exports :mod:`lzma`
> 
> * :mod:`!compression.zlib` -- Re-exports :mod:`zlib`
> 
> * :mod:`compression.zstd` -- Wrapper for the Zstandard compression library

Reviewer 2: ```suggestion
* :mod:`!compression.bz2` -- Re-exports :mod:`bz2`
* :mod:`!compression.gzip` -- Re-exports :mod:`gzip`
* :mod:`!compression.lzma` -- Re-exports :mod:`lzma`
* :mod:`!compression.zlib` -- Re-exports :mod:`zlib`
* :mod:`compression.zstd` -- Wrapper for the Zstandard compression library
```

> 

Reviewer 2: ```suggestion
```
